### ISSUE TITLE
panic while calling Error()

### ISSUE THREAD

**Issue by Termina1** [bytedance/sonic#700]
Found degradation in version v1.12.1, occurs when I'm trying to deserialise value that is number into string.

```
goroutine 9078663823 [running]:
github.com/bytedance/sonic/internal/decoder/errors.swithchJSONType(...)
	/go/pkg/mod/github.com/bytedance/sonic@v1.12.3/internal/decoder/errors/errors.go:145
github.com/bytedance/sonic/internal/decoder/errors.MismatchTypeError.Error({0x3, {0xc038c962d8, 0x3}, {0x2fb2530, 0x2057200}})
	/go/pkg/mod/github.com/bytedance/sonic@v1.12.3/internal/decoder/errors/errors.go:162 +0x24e
```

**Comment by liuq19** [comment]
please update sonic version into v1.12.2

**Comment by Termina1** [comment]
This reproduces on the latest version of sonic

**Comment by liuq19** [comment]
Are there any reproducible examples, with a complete panic stack and `go env`?

**Comment by liuq19** [comment]
any updates? @Termina1

**Comment by liuq19** [comment]
Fixed by #717
